src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -916,6 +916,9 @@ - (void)invalidate
         // the subscrption after this point.
         std::lock_guard lock(self->_lock);
         self->_currentReadClient = nullptr;
```

Comment:
Does this have the same problem as below?

Reply:
Yes this is an existing issue that invalidate doesn't actually delete the SubscriptionCallback object.

Reply:
And ReadClient is held by SubscriptionCallback, and the callback object destructor nils / deletes the ReadClient.